hooks/terraform_validate.sh

```diff
@@ -43,16 +43,88 @@ function per_dir_hook_unique_part {
   local -a -r args=("$@")
 
   local exit_code
-  local validate_output
+  #
+  # Get hook settings
+  #
+  local retry_once_with_cleanup=false
+
+  IFS=";" read -r -a configs <<< "${HOOK_CONFIG[*]}"
+
+  for c in "${configs[@]}"; do
+
+    IFS="=" read -r -a config <<< "$c"
+    key=${config[0]}
+    value=${config[1]}
+
+    case $key in
+      --retry-once-with-cleanup)
+        retry_once_with_cleanup=$value
+        ;;
+    esac
+  done
 
   common::terraform_init 'terraform validate' "$dir_path" || {
     exit_code=$?
     return $exit_code
   }
 
-  # pass the arguments to hook
-  validate_output=$(terraform validate "${args[@]}" 2>&1)
-  exit_code=$?
+  function do_validate {
+    validate_output=$(terraform validate "${args[@]}" 2>&1)
+    exit_code=$?
+    return $exit_code
+  }
+
+  function parse_validate {
+    # Requires jq
+    local exit_code
+    local validate_output
+    local valid
+    local summary
+
+    validate_output=$(terraform validate -json "${args[@]}" 2>&1)
+    exit_code=$?
+
+    valid="$(jq -rc '.valid' <<< "$validate_output")"
+
+    if [ "$valid" == "true" ]; then
+      return 0
+    fi
+
+    # Pretty-print error information
+    echo "$validate_output" | jq '.diagnostics[]'
+
+    # Parse error message
+    while IFS= read -r error_message; do
+      summary="$(jq -rc '.summary' <<< "$error_message")"
+      case $summary in
+        "missing or corrupted provider plugins")
+          return 10
+          ;;
+        "Module source has changed")
+          return 10
+          ;;
+      esac
+    done < <(jq -rc '.diagnostics[]' <<< "$validate_output")
+    return $exit_code
```

Comment:
Would it make sense to add an explanatory comment here?
```suggestion

    # Return `terraform validate`'s original exit code for
    # when `$summary` isn't covered by `case` block above
    return $exit_code
```